diff --git a/src/responder.rs b/src/responder.rs
index becffc6..b794c3e 100644
--- a/src/responder.rs
+++ b/src/responder.rs
@@ -1,3 +1,5 @@
+use std::time::Duration;
+
 use reqwest::{blocking::{Client, Response}, header::{CONTENT_TYPE, HeaderMap}};
 use crate::utils::debug::{log_request, log_response};

@@ -38,6 +40,7 @@ pub fn simple_post(server_url: String, api_key: &str, body: String, content_type
     let client = Client::new();
     let response = client
         .post(&server_url)
+        .timeout(Duration::from_mins(10))
         .headers(headers)
         .body(body)
         .query(&query)
@@ -52,4 +55,4 @@ pub fn simple_post(server_url: String, api_key: &str, body: String, content_type
         }
         std::process::exit(1);
     }
-}
\ No newline at end of file
+}
